src/language-js/embed.js

```diff
@@ -648,4 +652,61 @@ function printHtmlTemplateLiteral(path, print, textToDoc, parser, options) {
   );
 }
 
+/*
+ * ember-cli-htmlbars
+ * hbs`...`
+ * Handlebars comment block
+ */
+function isHandlebars(path) {
+  const node = path.getValue();
+  const parent = path.getParentNode();
+
+  return (
+    hasLanguageComment(node, "hbs") ||
+    (parent &&
+      parent.type === "TaggedTemplateExpression" &&
+      parent.tag.type === "Identifier" &&
+      parent.tag.name === "hbs")
+  );
+}
+
+function printHandlebarsTemplateLiteral(path, print, textToDoc, options) {
+  const node = path.getValue();
+
+  const text = node.quasis.map((quasi) => quasi.value.cooked).join("");
+
+  const expressionDocs = path.map(print, "expressions");
+
+  if (expressionDocs.length === 0 && text.trim().length === 0) {
+    return "``";
+  }
+
+  const contentDoc = textToDoc(text, { parser: "glimmer" });
```

Comment:
Can  we  call  `stripTrailingHardline`  here ? We  are  planning to  strip all  the finnal new line in  `textToDoc` function.

https://github.com/prettier/prettier/pull/8595


Just  a  question.

Reply:
Yes we can and we will be able to remove it after #8595. Interesting I didn't notice that discussion.

I'm adding it back.

Reply:
No need extra changes? Why finnal new line didn't effect.

Reply:
Because there is already no final newline in handlebars formatted templates.

A while ago, I wanted to change that (https://github.com/prettier/prettier/pull/6243). But it was a dead end.

Reply:
An option to change that could be to read the insert_final_newline option from .editorconfig. But it is not yet supported in the prettier package that reads editorconfig and I’m not sure it would be accepted?

Reply:
I don't know that, but we always insert one, maybe we should open an issue and discuss about that.

Reply:
Good to know, thanks.